packages/vue/test/errorHandler.test.ts

```diff
@@ -164,137 +170,102 @@ describe('attachErrorHandler', () => {
 
         // assert
         t.expect.errorHandlerSpy.toHaveBeenCalledWith(expect.any(Error), vm, 'stub-lifecycle-hook');
+        t.expect.errorToHaveBeenCaptured().withMechanismMetadata({ handled: true, type: 'vue-errorHandler' });
       });
     });
   });
 
-  describe('error logging', () => {
-    describe('given I disabled error logging', () => {
-      describe('when an error is captured', () => {
-        test('it logs nothing', () => {
-          // arrange
-          const vm = {
-            $options: {
-              name: 'stub-vm',
-            },
-          };
-          const t = testHarness({
-            enableWarnHandler: false,
-            logErrors: false,
-            vm,
-          });
-
-          // act
-          t.run();
+  describe('error re-throwing and logging', () => {
+    afterEach(() => {
+      vi.resetAllMocks();
+    });
 
-          // assert
-          t.expect.consoleErrorSpy.not.toHaveBeenCalled();
-          t.expect.warnHandlerSpy.not.toHaveBeenCalled();
+    describe('error re-throwing', () => {
+      it('should re-throw error when no error handler exists', () => {
+        const t = testHarness({
+          enableErrorHandler: false,
+          enableConsole: true,
+          vm: { $options: { name: 'stub-vm' } },
         });
+
+        expect(() => t.run()).toThrow(DummyError);
       });
-    });
 
-    describe('given I enabled error logging', () => {
-      describe('when I provide a `warnHandler`', () => {
-        describe('when a error is captured', () => {
-          test.each([
-            [
-              'with wm',
-              {
-                $options: {
-                  name: 'stub-vm',
-                },
-              },
-              generateComponentTrace({
-                $options: {
-                  name: 'stub-vm',
-                },
-              } as ViewModel),
-            ],
-            ['without vm', null, ''],
-          ])('it calls my `warnHandler` (%s)', (_, vm, expectedTrace) => {
-            // arrange
-            const t = testHarness({
-              vm,
-              logErrors: true,
-              enableWarnHandler: true,
-            });
+      it('should call user-defined error handler when provided', () => {
+        const vm = { $options: { name: 'stub-vm' } };
+        const t = testHarness({
+          enableErrorHandler: true,
+          enableConsole: true,
+          vm,
+        });
 
-            // act
-            t.run();
+        t.run();
 
-            // assert
-            t.expect.consoleErrorSpy.not.toHaveBeenCalled();
-            t.expect.warnHandlerSpy.toHaveBeenCalledWith(
-              'Error in stub-lifecycle-hook: "DummyError: just an error"',
-              vm,
-              expectedTrace,
-            );
-          });
-        });
+        t.expect.errorHandlerSpy.toHaveBeenCalledWith(expect.any(Error), vm, 'stub-lifecycle-hook');
```

Comment:
not sure if we already test this in the harness somehow but are we asserting that we don't rethrow in case users declare their errorHandler?

Reply:
thx for explaining!